src/components/WeeklySummary/WriteItForMeModal.jsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from 'react';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import { boxStyle } from 'styles';
+import httpService from 'services/httpService';
+import { connect } from 'react-redux';
+import { ENDPOINTS } from '../../utils/URL';
+import { SET_CURRENT_USER } from '../../constants/auth';
+import { getUserInfo } from '../../utils/permissions';
+
+function WriteItForMeModal(props) {
+  const { pasteResponse } = props;
+  const [modal, setModal] = useState(false);
+  const [summary, setSummary] = useState();
+  const toggle = () => setModal(!modal);
+
+  const fetchSummary = async () => {
+    // eslint-disable-next-line react/destructuring-assignment
+    const { userid } = getUserInfo();
+    const { displayUserProfile } = props;
+    toggle();
+
+    httpService
+      .post(ENDPOINTS.INTERACT_WITH_CHATGPT, {
+        userid,
```

Comment:
I looked at the requested payload data. The only data being passed was the name of the user. I did not see userid. Was this behavior happening on your end?